runner/output.go

```diff
@@ -26,4 +26,9 @@ type Output interface {
 
 	// Copies (if necessary) the URI target to local file and returns that absolute file path.
 	AsFile() string
+
+	// Add this Output to a list, probably an endpoints.ResourceHandler.
+	// Multiple names may be registered under a single namespace.
+	// Returns the uri resulting from this registration.
+	Register(namespace, name string) (uri string)
```

Comment:
Could you help me understand with examples?

I think it'd be nice to have less coupling between these structures. I'd hope that output.go doesn't have to know about ResourceHandler *at all*.